d2bs/kolbot/libs/bots/BoBot.js

```diff
@@ -0,0 +1,230 @@
+/**
+*	@filename	BoBot.js
+*	@author		casualBotter
+*	@desc		Meant to run alongside Enchant to Bo players
+*/
+
+function BoBot(){
+	var command, hostile, nick, spot, tick, s, m,
+		startTime = getTickCount(),
+		shitList = [],
+		greet = [];
+
+		this.Bo = function (nick) {
+		if (!Misc.inMyParty(nick)) {
+			say("Accept party invite, noob.");
+
+			return false;
+		}
+
+		var partyUnit,
+			unit = getUnit(0, nick);
+
+		if (getDistance(me, unit) > 10) {
+			say("Get closer.");
+
+			return false;
+		}

```

Comment:
Isn't the unit supposed to be waiting at a wp and not in town ?

Reply:
It waits in town until called, I am having trouble if someone calls bo and they aren't right on the wp it says they are too far. Any ideas on how to fix that i've been trying to alter the range for getunit but its not helping

Reply:
The unit is the player, it should probably be waiting for you near a waypoint so you won't be able to use `getUnit()` which has a limited range.

You could remove this part, keep `getParty()` and make sure the player's area has a waypoint and is not a town.